dev/com.ibm.ws.sipcontainer/src/com/ibm/ws/sip/container/servlets/SipApplicationSessionImpl.java

```diff
@@ -1800,4 +1806,36 @@ public void setSessionKeyBaseTargeting(String skbt){
 		}
 		m_sessionKeyBaseKey = skbt;
 	}
+	
+	public Iterator getSessions(String protocol, boolean b) {
+		if (protocol.equalsIgnoreCase("SIP"))  {  //get SIP application sessions  
+			if (c_logger.isTraceDebugEnabled()) {
+				c_logger.traceDebug("getSessions(" + protocol + ", " + b + " detected.");
+			}
+			
+			if (b) {  //boolean is true	
+					return (Iterator)getAllSIPSessions(true).iterator();
+				}
+		
+			else {  //boolean is false
+					return (Iterator)getAllSIPSessions(false).iterator();
+				}
+		}
+		
+		else if (protocol.equalsIgnoreCase("HTTP")){
+			return Collections.EMPTY_MAP.keySet().iterator();
+		}
+			
+		else { 
+
+			SipAppDesc sipAppDesc = getAppDescriptor();
+			if(sipAppDesc != null && !sipAppDesc.isJSR289Application()){
```

Comment:
I'm not familiar with this. What does this mean?  I guess what I mean under what condition would you not throw the IllegalArgumentException?

Reply:
sipcontainer only knows about SIP and HTTP protocols. We are not restricting what string will be passed into this getSessions(String, boolean) method, but we are only handling SIP and HTTP. If anything else is passed in this method, the IllegalArgumentException will be throws. This is the same in the previously existing getSessions(String) method.

Reply:
I guess what's confusing me here is that you can pass in a bad protocol, but it should still return a iterator here because of `sipAppDesc != null && !sipAppDesc.isJSR289Application()` .

Reply:
it is the same logic we had in place all along. I don't think we should modify it without a good reason.